tools/run_tests/run_xds_tests.py

```diff
@@ -499,6 +535,93 @@ def test_secondary_locality_gets_requests_on_primary_failure(
         patch_backend_instances(gcp, backend_service, [primary_instance_group])
 
 
+def test_traffic_splitting(gcp, original_backend_service, instance_group,
+                           alternate_backend_service, same_zone_instance_group):
+    # This test start with all traffic going to original_backend_service. Then
+    # it updates URL-map to set default action to traffic splitting between
+    # original and alternate. It waits for all backends in both services to
+    # receive traffic, then verifies that weights are expected.
+    logger.info('Running test_traffic_splitting')
+
+    logger.info('waiting for original to become healthy')
+    wait_for_healthy_backends(gcp, original_backend_service, instance_group)
+
+    patch_backend_instances(gcp, alternate_backend_service,
+                            [same_zone_instance_group])
+    logger.info('waiting for alternate to become healthy')
+    wait_for_healthy_backends(gcp, alternate_backend_service,
+                              same_zone_instance_group)
+
+    original_backend_instances = get_instance_names(gcp, instance_group)
+    logger.info('original backends instances: %s', original_backend_instances)
+
+    alternate_backend_instances = get_instance_names(gcp,
+                                                     same_zone_instance_group)
+    logger.info('alternate backends instances: %s', alternate_backend_instances)
+
+    # Start with all traffic going to original_backend_service.
+    logger.info('waiting for traffic to all go to original')
+    wait_until_all_rpcs_go_to_given_backends(original_backend_instances,
+                                             _WAIT_FOR_STATS_SEC)
+
+    try:
+        # Path urlmap, change route action to traffic splitting between original
+        # and alternate.
+        logger.info('patching url map with traffic splitting')
+        expected_service_percentage = [20, 80]
+        patch_url_map_weighted_backend_services(
+            gcp, {
+                original_backend_service: expected_service_percentage[0],
+                alternate_backend_service: expected_service_percentage[1],
+            })
+        expected_instance_percentage = [
+            expected_service_percentage[0] * 1.0 /
+            len(original_backend_instances)
+        ] * len(original_backend_instances) + [
+            expected_service_percentage[1] * 1.0 /
+            len(alternate_backend_instances)
+        ] * len(alternate_backend_instances)
+
+        # Wait for traffic to go to both services.
+        logger.info(
+            'waiting for traffic to go to all backends (including alternate)')
+        wait_until_all_rpcs_go_to_given_backends(
+            original_backend_instances + alternate_backend_instances,
+            _WAIT_FOR_STATS_SEC)
+
+        # Verify that weights between two services are expected.
+        retry_count = 3
```

Comment:
This means that the updated weights must be sent to the client within ~30 seconds, right (each attempt waits for # RPCs = QPS * 10) That might be a bit tight given experience with latency for other updates to propagate from GCE to TD to clients.

Reply:
Changed to 10 tries, that's about 100 seconds. Would it be long enough?

Reply:
Based on known config update latencies, yes that should be enough